test/org/sagebionetworks/bridge/upload/UploadUtilTest.java

```diff
@@ -588,4 +645,19 @@ public void iosTimestamp() {
         DateTime parsedTimestamp = UploadUtil.parseIosTimestamp("2015-08-26 16:54:04 -0700");
         assertEquals(expectedMillis, parsedTimestamp.getMillis());
     }
+
+    @Test
+    public void sanitizeFieldNames() {
+        // Test cases: map with 2 fields, one of which needs to be sanitized.
+        // Use an immutable map. If we attempt to modify the original map, this will throw.
+        // Use a map of strings, for simplicity of testing.
+        Map<String, String> inputMap = ImmutableMap.<String, String>builder().put("foo", "bar")
+                .put("sanitize!@#$this", "sanitize this's value").build();
+
+        // Execute and validate.
+        Map<String, String> outputMap = UploadUtil.sanitizeFieldNames(inputMap);
+        assertEquals(2, outputMap.size());
+        assertEquals("bar", outputMap.get("foo"));
+        assertEquals("sanitize this's value", outputMap.get("sanitize____this"));
```

Comment:
Is there any danger of two strings like "a!@#b" and "a^&*b" both being sanitized to a___b and a___b? If so, should you opt for some kind of escaping of special characters so the fields remain unique?

Reply:
The limitation is that you can't create Synapse columns with these characters, and we enforce that same limitation in schema fields. There would be no point trying to preserve these special characters in field names, since we can't use them anyway.

Reply:
Not preserve but escape, so for example using URL escaping, "a!%40%23b" and "a%5E%26*b", which remain different but also safe for use. However, changes of this being an issue seem very slim.